test/functional/rpc_rawtransaction.py

```diff
@@ -62,44 +62,51 @@ def set_test_params(self):
         # whitelist all peers to speed up tx relay / mempool sync
         for args in self.extra_args:
             args.append("-whitelist=noban@127.0.0.1")
+        if self.is_wallet_compiled():
+            self.requires_wallet = True
 
         self.supports_cli = False
 
     def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
+        if self.is_wallet_compiled():
+            if self.options.descriptors:
+                self.skip_if_no_sqlite()
+            else:
+                self.skip_if_no_bdb()
 
     def setup_network(self):
         super().setup_network()
         self.connect_nodes(0, 2)
 
     def run_test(self):
+        self.wallet = MiniWallet(self.nodes[0])
         self.log.info("Prepare some coins for multiple *rawtransaction commands")
-        self.generate(self.nodes[2], 1)
+        self.generate(self.wallet, 10)
         self.generate(self.nodes[0], COINBASE_MATURITY + 1)
-        for amount in [1.5, 1.0, 5.0]:
-            self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), amount)
-        self.sync_all()
-        self.generate(self.nodes[0], 5)
+        if self.is_wallet_compiled():
```

Comment:
```suggestion
        if self.requires_wallet:
```

maybe you can avoid calls to `is_wallet_compiled()` since you defined a var for it in `set_test_params`:
```py
if self.is_wallet_compiled():
    self.requires_wallet = True
```

Reply:
Oh, right! I just updated it, thank you :)